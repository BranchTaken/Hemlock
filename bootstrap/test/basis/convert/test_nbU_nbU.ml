open! Basis.Rudiments
open! Basis
open Format

let test () =
  printf "@[<h>";
  List.iter U8.([kv 0L; kv 1L; kv 127L; kv 128L; kv 255L]) ~f:(fun u ->
    printf "extend_to_u32 %a -> %a\n"
      U8.xpp u
      U32.xpp (U8.extend_to_u32 u)
  );
  printf "\n";
  List.iter U32.([kv 0L; kv 1L; kv 255L; kv 256L; kv 511L]) ~f:(fun u ->
    printf "trunc_of_u32/narrow_of_u32_opt %a -> %a/%a\n"
      U32.xpp u
      U8.xpp (U8.trunc_of_u32 u)
      (Option.xpp U8.xpp) (U8.narrow_of_u32_opt u)
  );
  printf "@]"

let _ = test ()
