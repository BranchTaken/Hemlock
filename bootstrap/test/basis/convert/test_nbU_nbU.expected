extend_to_u32 0u8 -> 0u32
extend_to_u32 1u8 -> 1u32
extend_to_u32 127u8 -> 127u32
extend_to_u32 128u8 -> 128u32
extend_to_u32 255u8 -> 255u32

trunc_of_u32/narrow_of_u32_opt 0u32 -> 0u8/Some 0u8
trunc_of_u32/narrow_of_u32_opt 1u32 -> 1u8/Some 1u8
trunc_of_u32/narrow_of_u32_opt 255u32 -> 255u8/Some 255u8
trunc_of_u32/narrow_of_u32_opt 256u32 -> 0u8/None
trunc_of_u32/narrow_of_u32_opt 511u32 -> 255u8/None
